src/service/chat.js

```diff
@@ -203,6 +203,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPost` function seems to be handling the API call and error handling correctly. However, it's not clear what happens when the `source` parameter is neither 'post' nor 'comment'. If this function should only handle these two cases, consider throwing an error in the default case of the switch statement to prevent misuse of the function.
```diff
    default:
+       throw new Error(`Invalid source: ${source}`);
      break;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->